tests/state_test_util.go

```diff
@@ -146,7 +146,16 @@ func (t *StateTest) Run(subtest StateSubtest, vmconfig vm.Config) (*state.StateD
 	if logs := rlpHash(statedb.Logs()); logs != common.Hash(post.Logs) {
 		return statedb, fmt.Errorf("post state logs hash mismatch: got %x, want %x", logs, post.Logs)
 	}
-	root, _ := statedb.Commit(config.IsEIP158(block.Number()))
+	// Commit block
+	statedb.Commit(config.IsEIP158(block.Number()))
+	// Add 0-value mining reward. This only makes a difference in the cases
+	// where
+	// - the coinbase suicided, or
+	// - there are only 'bad' transactions, which aren't executed. In those cases,
+	//   the coinbase gets no txfee, so isn't created, and thus needs to be touched
+	statedb.AddBalance(block.Coinbase(), new(big.Int))
```

Comment:
But why does the coinbase get tx fees in state tests?

Reply:
Yeah. There's a discussion around that, which I linked to within Gitter. There's no real reason for it, other than that @winsvega got that into Cpp, and that makes it easier to view a statetest as a mini-version of a blockchain test. 

I mean there's no reason as in 'It's correct according to the YP' - the reason is simply an implementation change in the test generator  process.   

@winsvega feel free to chime in

Reply:
Would there be a difference if you did `AddBalance` before `Commit`? It would simplify this change.

Reply:
Yes, because then the `touch` happens before suicides are handled, which is at the end of transaction processing. Thus, we need to commit once to run suicides, and only afterwards do the coinbase touch.

Reply:
Added some docs about that